assignments/lambda-classes.js

```diff
@@ -1 +1,211 @@
 // CODE here for your Lambda Classes
+
+/* 
+
+Roster of Lambda personnel:
+Lambda personnel can be broken down into three different types of people:
+Instructors - extensions of Person, Students - extensions of Person, Project Managers - extensions of Instructors
+
+*/
+
+class Person {
+  constructor(attributes) {
+    this.name = attributes.name;
+    this.age = attributes.age;
+    this.location = attributes.location;
+    this.gender = attributes.gender;
+  }
+  speak() {
+    console.log(`Hello my name is ${this.name}, I am from ${this.location}.`);
+  }
+}
+
+class Instructor extends Person {
+  constructor(attributes2) {
+    super(attributes2);
+    this.specialty = attributes2.specialty;
+    this.favLanguage = attributes2.favLanguage;
+    this.catchPhrase = attributes2.catchPhrase;
+  }
+  demo(subject) {
+    console.log(`Today we are learning about ${subject}`);
+  }
+  grade(student, subject) {
+    console.log(`${student.name} receives a perfect score on ${subject}`);
+  }
+}
+
+class Student extends Person {
+  constructor(attributes3) {
+    super(attributes3);
+    this.previousBackground = attributes3.previousBackground;
+    this.className = attributes3.className;
+    this.favSubjects = attributes3.favSubjects;
+    this.grade = attributes3.grade;
+  }
+  listsSubjects() {
+    this.favSubjects.forEach(element => {
```

Comment:
Going further, you might find switching forEach out for map to have higher performance and with regards to functional programming philosophy and the integrity of data in React, map will not alter the original array where for each mutates the original and points its reference to the result of forEach and this would cause breaks when you encounter list rendering in React.